src/metabase/driver/druid.clj

```diff
@@ -0,0 +1,192 @@
+  (let [[{{:keys [pagingIdentifiers events]} :result}] (do-query details {:queryType    :select
+                                                                          :dataSource   table-name
+                                                                          :intervals    ["-5000/5000"]
+                                                                          :granularity  :all
+                                                                          :dimensions   [field-name]
+                                                                          :metrics      []
+                                                                          :pagingSpec   (merge {:threshold driver/field-values-lazy-seq-chunk-size}
+                                                                                               (when paging-identifiers
+                                                                                                 {:pagingIdentifiers paging-identifiers}))})]
+    ;; return pair of [paging-identifiers values]
+    [ ;; Paging identifiers return the largest offset of their results, e.g. 49 for page 1.
+     ;; We need to inc that number so the next page starts after that (e.g. 50)
+     (let [[[k offset]] (seq pagingIdentifiers)]
+       {k (inc offset)})
+     ;; Unnest the values
+     (for [event events]
+       (get-in event [:event (keyword field-name)]))]))
+
+(defn- field-values-lazy-seq
+  ([_ field]
+   (field-values-lazy-seq (:details (u/deref-> field :table :db))
+                          (:name @(:table field))
+                          (:name field)
+                          0
+                          nil))
+
+  ([details table-name field-name total-items-fetched paging-identifiers]
+   {:pre [(map? details) (u/string-or-keyword? table-name) (u/string-or-keyword? field-name) (integer? total-items-fetched) (u/nil-or? map? paging-identifiers)]}
+   (lazy-seq (let [[paging-identifiers values] (field-values-lazy-seq-fetch-one-page details table-name field-name paging-identifiers)
+                   total-items-fetched         (+ total-items-fetched driver/field-values-lazy-seq-chunk-size)]
+               (concat values
+                       (when (and (seq values)
+                                  (< total-items-fetched driver/max-sync-lazy-seq-results)
+                                  (= (count values) driver/field-values-lazy-seq-chunk-size))
+                         (field-values-lazy-seq details table-name field-name total-items-fetched paging-identifiers)))))))
+
+
+;;; ### DriverDriver Class Definition
+
+(defrecord DruidDriver []
+  clojure.lang.Named
+  (getName [_] "Druid"))
+
+(extend DruidDriver
+  driver-util/IDriverTableToColumnNames
+  {:table->column-names table->column-names}
+
+  driver/IDriver
+  (merge driver/IDriverDefaultsMixin
+         {:active-column-names->type (fn [driver table]
```

Comment:
why is it necessary to use this approach for getting the column types?  i thought that Druid already has an api endpoint which provides the descriptions of the tables?